### PR TITLE
Update ulog2csv.py

### PR DESCRIPTION
**Pull request by tayyabkhalil-313**
I had a huge log file from a flight but the desired part of the log file only consisted of data towards the end of log file. So i added an additional optional parameter for the `ulog2csv` command to only convert data after the particular time. 

#### Usage:

`ulog2csv [-t time] file.ulg`
Convert the `file.ulg` to csv for which timestamp exceeds or equals `time` in seconds.

### CODE REVIEW

File: pyulog/ulog2csv.py [hunk 1]
```diff
@@ -31,25 +32,29 @@ def main():
                         help='Output directory (default is same as input file)',
                         metavar='DIR')
     parser.add_argument('-i', '--ignore', dest='ignore', action='store_true',
-                        help='Ignore string parsing exceptions', default=False)
-
+                        help='Ignore string parsing exceptions', default=False)               
+    parser.add_argument(
+        '-t', '--time_s', dest='time_s', type = int, 
+        help="Only convert data after this timestamp(in seconds)")
```
Comment (bkueng) [inline pyulog/ulog2csv.py:38]:
```suggestion
        help="Only convert data after this timestamp (in seconds)")
```

File: pyulog/ulog2csv.py [hunk 2]
```diff
@@ -31,25 +32,29 @@ def main():
                         help='Output directory (default is same as input file)',
                         metavar='DIR')
     parser.add_argument('-i', '--ignore', dest='ignore', action='store_true',
-                        help='Ignore string parsing exceptions', default=False)
-
+                        help='Ignore string parsing exceptions', default=False)               
+    parser.add_argument(
+        '-t', '--time_s', dest='time_s', type = int,
```
Comment (bkueng) [inline pyulog/ulog2csv.py:37]: Can you also add the time before parameter (`time_e`)?

File: pyulog/ulog2csv.py [hunk 3]
```diff
@@ -78,21 +83,22 @@ def convert_ulog2csv(ulog_file_name, messages, output, delimiter, disable_str_ex
 
             # use same field order as in the log, except for the timestamp
             data_keys = [f.field_name for f in d.field_data]
-            data_keys.remove('timestamp')
-            data_keys.insert(0, 'timestamp')  # we want timestamp at first position
-
+            data_keys.remove('timestamp')   
+            data_keys.insert(0, 'timestamp')  # we want timestamp at first positio
```
Comment (bkueng) [inline pyulog/ulog2csv.py:87]: Can you revert these whitespace changes?

**Review by bkueng — COMMENTED** [review]
Thanks, can you fix the formatting errors from ci?
```
pyulog/ulog2csv.py:35:79: C0303: Trailing whitespace (trailing-whitespace)
pyulog/ulog2csv.py:39:0: C0303: Trailing whitespace (trailing-whitespace)
pyulog/ulog2csv.py:46:0: C0301: Line too long (105/100) (line-too-long)
pyulog/ulog2csv.py:49:0: C0301: Line too long (104/100) (line-too-long)
pyulog/ulog2csv.py:86:41: C0303: Trailing whitespace (trailing-whitespace)
pyulog/ulog2csv.py:104:0: C0303: Trailing whitespace (trailing-whitespace)
```

File: pyulog/ulog2csv.py (hunk 2 above)
Comment (tayyabkhalil-313) [reply pyulog/ulog2csv.py:38]: What do you mean exactly?

Comment (bkueng) [reply pyulog/ulog2csv.py:38]: You added a parameter to convert after a certain timestamp. The opposite makes sense as well: convert up to a certain timestamp. So I would add both options.

File: test/test_ulog2csv.py [hunk 4]
```diff
@@ -46,7 +46,9 @@ def test_ulog2csv(self, test_case):
         messages = []
         output=tmpdir
         delimiter=','
-        ulog2csv.convert_ulog2csv(ulog_file_name, messages, output, delimiter)
+        time_s = 0
+        time_e = 1
```
Comment (bkueng) [inline test/test_ulog2csv.py:50]: The unit test is failing, problably best to set this to 0.

```suggestion
        time_e = 0
```

```
                #get the index for row where timestamp exceeds or equals the required value
                time_s_i = np.where(d.data['timestamp'] >= time_s * 1e6)[0][0] if time_s else 0
                #get the index for row upto the timestamp of the required value
>               time_e_i = np.where(d.data['timestamp'] >= time_e * 1e6)[0][0] if time_e else len(d.data['timestamp'])
E               IndexError: index 0 is out of bounds for axis 0 with size 0
```

**Review by bkueng — APPROVED** [review]

**Comment by augustonetto** [comment]
Hi!

For some reason this commit broke my python script and the error message is not helping. As this is an option, I believe nothing should be changed if I don't pass the arguments to the function, no?

This was working: 

`ulog2csv.convert_ulog2csv(ulog_file_name=file_ulg, messages='', output=outputPath, delimiter=',')`

Now the error is: 
`Python Error: TypeError: convert_ulog2csv() missing 2 required positional arguments: 'time_s' and 'time_e'`

How to modify it to work again?

Thanks :)

**Comment by bkueng** [comment]
You can use `ulog2csv.convert_ulog2csv(ulog_file_name=file_ulg, messages='', output=outputPath, delimiter=',', time_s=None, time_e=None)`

**Pull request merged**